fiftyone/core/state.py

```diff
@@ -149,6 +149,11 @@ def from_dict(cls, d, with_config=None):
 
         fo.config.timezone = d.get("config", {}).get("timezone", None)
 
+        spaces = d.get("spaces", None)
+
+        if spaces is not None:
+            spaces = Space.from_json(spaces)
```

Comment:
I added this so that `session.spaces` was available as a `Space` object rather than a string in Python

Reply:
This is fixed in newest commit of PR https://github.com/voxel51/fiftyone/pull/2510
It's a  little different. `spaces` is coming as a string so I had to parse to JSON first